scripts/install-helpers/baremetal-coco/install.sh

```diff
@@ -0,0 +1,733 @@
+    echo "GA_RELEASE: $GA_RELEASE"
+    echo "MIRRORING: $MIRRORING"
+    echo "TEE_TYPE: $TEE_TYPE"
+    echo "SKIP_NFD: $SKIP_NFD"
+    echo "TRUSTEE_URL: $TRUSTEE_URL"
+}
+
+while getopts "t:hmsbu" opt; do
+    case $opt in
+    t)
+        # Convert it to lower case
+        TEE_TYPE=$(echo "$OPTARG" | tr '[:upper:]' '[:lower:]')
+        echo "TEE type: $TEE_TYPE"
+        ;;
+    h)
+        display_help
+        exit 0
+        ;;
+    m)
+        echo "Mirroring option passed"
+        # Set global variable to indicate mirroring option is passed
+        MIRRORING=true
+        ;;
+    s)
+        echo "Setting additional cluster-wide image pull secret"
+        ADD_IMAGE_PULL_SECRET=true
+        ;;
+    b)
+        echo "Using non-ga operator bundles"
+        GA_RELEASE=false
+        ;;
+    u)
+
+        ADD_IMAGE_PULL_SECRET=false
+        # Ensure TEE_TYPE is set
+        verify_params
+        echo "Uninstalling"
+        uninstall
+        exit 0
+        ;;
+
+    \?)
+        echo "Invalid option: -$OPTARG" >&2
+        display_help
+        exit 1
+        ;;
+    esac
+done
+
+# Verify all required parameters are set
+verify_params
+
+# Check if oc command is available
+check_command "oc"
+
+# Display the cluster information
+oc cluster-info || exit 1
+
+# If MIRRORING is true, then create the image mirroring config
+if [ "$MIRRORING" = true ]; then
+    echo "Creating image mirroring config"
+    oc apply -f image_mirroring.yaml || exit 1
+
+    # Sleep for sometime before checking MCP status
+    sleep 10
+
+    echo "Waiting for MCP to be ready"
+    wait_for_mcp master || exit 1
+    wait_for_mcp worker || exit 1
+fi
+
+# If ADD_IMAGE_PULL_SECRET is true, then add additional cluster-wide image pull secret
+if [ "$ADD_IMAGE_PULL_SECRET" = true ]; then
+    echo "Adding additional cluster-wide image pull secret"
+    # Check if jq command is available
+    check_command "jq"
+    add_image_pull_secret
+
+    # Sleep for sometime before checking MCP status
+    sleep 10
+
+    echo "Waiting for MCP to be ready"
+    wait_for_mcp master || exit 1
+    wait_for_mcp worker || exit 1
+
+fi
+
+# Deploy NFD operator and create NFD CR if SKIP_NFD is false
+if [ "$SKIP_NFD" = false ]; then
+    deploy_nfd_operator || exit 1
+
+    # Create NFD CR
+    oc apply -f nfd/nfd-cr.yaml || exit 1
+
+    case $TEE_TYPE in
+    tdx)
+        create_intel_node_feature_rules || exit 1
+        deploy_intel_device_plugins || exit 1
+        ;;
+    snp)
+        create_amd_node_feature_rules || exit 1
+        ;;
+    esac
+fi
+
+deploy_osc_operator || exit 1
+
+# Create CoCo feature gate ConfigMap
+oc apply -f osc-fg-cm.yaml || exit 1
+
+# Create Layered Image FG ConfigMap
+if [ "$TEE_TYPE" = "tdx" ]; then
+    oc apply -f layeredimage-cm-tdx.yaml || exit 1
+elif [ "$TEE_TYPE" = "snp" ]; then
+    oc apply -f layeredimage-cm-snp.yaml || exit 1
+else
+    echo "Unsupported TEE_TYPE. It must be tdx or snp" || exit 1
+fi
```

Comment:
`TEE_TYPE` is already validated by `verify_params()`. You can just use a `case` command as already done a few lines above.

```suggestion
    case $TEE_TYPE in
    tdx)
        oc apply -f layeredimage-cm-tdx.yaml || exit 1
        ;;
    snp)
        oc apply -f layeredimage-cm-snp.yaml || exit 1
        ;;
    esac
```